nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/probes/ProbeInfoWriter.java

```diff
@@ -0,0 +1,90 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.probes;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+
+import java.io.IOException;
+
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.core.management.api.ProbeInfo;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class ProbeInfoWriter extends AbstractJsonWriter<ProbeInfo> {
+
+    public static final String WARNINGS = "warnings";
+
+    public static final String WARNING = "warning";
+
+    public static final String NAME = "name";
+
+    public static final String VERSION = "version";
+
+    @Override
+    public void write(ProbeInfo entity, JsonGenerator jg) throws IOException {
+        jg.useDefaultPrettyPrinter();
+        jg.writeStartObject();
+
+        jg.writeStringField("shortcutName", entity.getShortcutName());
+        jg.writeStringField("success", String.valueOf(entity.getStatus().isSuccess()));
+        jg.writeStringField("result", entity.getStatus().toString());
+
+        jg.writeArrayFieldStart("history");
+
+        String value = (entity.getLastRunnedDate() == null ) ? null : entity.getLastRunnedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last run", value);
+        jg.writeEndObject();
+
+        value = (entity.getLastSucceedDate() == null ) ? null : entity.getLastSucceedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last success", value);
+        jg.writeEndObject();
+
+        value = (entity.getLastSucceedDate() == null ) ? null : entity.getLastFailedDate().toString();
+        jg.writeStartObject();
+        jg.writeStringField("last fail", value);
+        jg.writeEndObject();
+
+        jg.writeEndArray();
+        jg.writeArrayFieldStart("counts");
+
+        jg.writeStartObject();
+        jg.writeStringField("run count", String.valueOf(entity.getRunnedCount()));
+        jg.writeEndObject();
+
+        jg.writeStartObject();
+        jg.writeStringField("success count", String.valueOf(entity.getSucceedCount()));
+        jg.writeEndObject();
+
+        jg.writeStartObject();
+        jg.writeStringField("failure count", String.valueOf(entity.getFailedCount()));
+        jg.writeEndObject();
+        jg.writeEndArray();
+
+        jg.writeStringField("time", String.valueOf(entity.getLastDuration()));
```

Comment:
Same question about format + we can serialize a real java duration containing the format (seconds, millis...), but should be discussed with team first.

Reply:
Ok !

Reply:
what about using date in the ISO format, in Nuxeo side there is some examples; have look to [here](https://github.com/nuxeo/nuxeo/blob/7e61b0f0706e2974ce5e157aefb61eea1d17eee2/nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentModelJsonWriter.java#L166)

Reply:
the DateUtils class is in `nuxeo-commons` i think we can addd the maven dependency to the `nuxeo-management-rest-api`, of course to validate with the rest of the team. @ataillefer, @kevinleturc

Reply:
Adding `nuxeo-commons` is not an issue, it should already be in the classpath. @RSalem07, here it's a duration and not a date, we can serialize it as a number but we need to document somewhere the unit or we can serialize it as `Duration` with the JDK format or a 'custom' one like it was done in [NXP-25402](https://jira.nuxeo.com/browse/NXP-25402)

Reply:
I don't have DateUtils. Was it introduced in 11.1 ? I'm on 10.10

Reply:
It is introduced in 11.1 indeed.

You can take "idea" of class, see https://github.com/nuxeo/nuxeo/blob/master/nuxeo-common/src/main/java/org/nuxeo/common/utils/DateUtils.java#L49 to get pattern (without optional part) and the rest of class to get how to use java.time for formatting.

When we'll do the merge to master, we should replace this formatting by `DateUtils`. Maybe it worth to write it somewhere, probably in the merge ticket?